sdks/java/testing/tpcds/src/main/java/org/apache/beam/sdk/tpcds/SqlTransformRunner.java

```diff
@@ -306,6 +327,68 @@ public static void runUsingSqlTransform(String[] args) throws Exception {
 
     executor.shutdown();
 
-    printExecutionSummary(completion, queryNames.length);
+    List<TpcdsRunResult> results = collectTpcdsResults(completion, queryNames.length);
+
+    if (tpcdsOptions.getExportSummaryToInfluxDB()) {
+      final long timestamp = start.getMillis() / 1000; // seconds
+      savePerfsToInfluxDB(tpcdsOptions, results, timestamp);
+    }
+
+    printExecutionSummary(results);
+  }
+
+  private static void savePerfsToInfluxDB(
+      final TpcdsOptions options, final List<TpcdsRunResult> results, final long timestamp) {
+    final InfluxDBSettings settings = getInfluxSettings(options);
+    final Map<String, String> tags = options.getInfluxTags();
+    final String runner = options.getRunner().getSimpleName();
+    final List<Map<String, Object>> schemaResults =
+        results.stream()
+            .map(
+                entry ->
+                    getResultsFromSchema(
+                        entry, timestamp, runner, produceMeasurement(options, entry)))
+            .collect(toList());
+    InfluxDBPublisher.publishTpcdsResults(schemaResults, settings, tags);
+  }
+
+  private static String produceMeasurement(final TpcdsOptions options, TpcdsRunResult entry) {
+    return String.format(
+        "%s_%s_%s_%s",
+        options.getBaseInfluxMeasurement(),
+        entry.getQueryName(),
+        entry.getDialect(),
+        entry.getDataSize());
+  }
+
+  private static Map<String, Object> getResultsFromSchema(
+      final TpcdsRunResult results,
+      final long timestamp,
+      final String runner,
+      final String measurement) {
+    final Map<String, Object> schemaResults = new HashMap<>();
+    schemaResults.put("timestamp", timestamp);
+    schemaResults.put("runner", runner);
+    schemaResults.put("measurement", measurement);
+
+    // By default, InfluxDB treats all number values as floats. We need to add 'i' suffix to
+    // interpret the value as an integer.
+    final int runtimeMs =
+        results.getIsSuccessful()
+            ? (int) (results.getElapsedTime() * 1000)
+            : // change sec to ms
+            0;
+    schemaResults.put("runtimeMs", runtimeMs + "i");
```

Comment:
As mentioned above, I think converting datatypes for influx DB in this place smells a bit :/